packages/flutter/lib/src/widgets/image.dart

```diff
@@ -1299,6 +1299,29 @@ class _ImageState extends State<Image> with WidgetsBindingObserver {
       _completerHandle = _imageStream!.completer!.keepAlive();
     }
 
+    // It's almost time to remove the last listener, which triggers the
+    // disposal. But before that, add an ephemeral listener to potentially
+    // suppress errors.
+    //
+    // Reason: When an app provides an `Image` widget with an `errorBuilder`, it
+    // expects the widget to never report errors through `FlutterError` in any
+    // cases. This is hard if the stream fails after the disposal, because an
+    // image stream must have no listeners to be disposed, which then has
+    // nothing to suppress the errors. This is solve with the help of an
+    // ephemeral listener, which also suppresses the error but does not hinder
+    // disposal. For more details, see
+    // https://github.com/flutter/flutter/issues/97077 .
+    if (widget.errorBuilder != null) {
+      if (_imageStream!.completer == null) {
+        _imageStream!.setCompleter(_EmptyImageCompleter());
```

Comment:
If there is no completer, adding this won't do anything right? since it will never complete, should we just return in this case?

I looked at the code, and it looked like that even we set this completer and addEphemeralErrorListener, the EphemeralErrorListener will not carry over if the completer change later, so it won't be able to mute the error after that.

Reply:
> Do you mean that a stream without a completer will not cause any error, and it isn't useful to add ephemeral listeners any more?

technically yes.

but i am more focusing on that adding an empty completer to add a ephemeral listener seems to be the same as not adding an ephemeral listener at all. The empty completer is never going to throw, and changing the completer won't carry the ephemeral listener over.

As for whether it should mute the error throw after changing the completer if widget.errorbuilder != null, I am not sure what the correct behavior should be.

Reply:
> but i am more focusing on that adding an empty completer to add a ephemeral listener seems to be the same as not adding an ephemeral listener at all. The empty completer is never going to throw, and changing the completer won't carry the ephemeral listener over.

That makes sense. I'll look more into it and remove the empty listener if that's the case.

> As for whether it should mute the error throw after changing the completer if widget.errorbuilder != null, I am not sure what the correct behavior should be.

For now it will mute the error again. It simply takes whatever case at the moment of widget disposal.

Reply:
The code seems to show that errors are only triggered from within the completer. Therefore when there's no completers there is no need to create one. I've removed the creation.